### PR TITLE
Fix new Zones appearing as parent of a ghost child

### PR DESCRIPTION
**Pull request by AleziaKurdis**
Fix for Issue #869

This excludes the local and avatar entities from what is returns by Entities.getChildrenIDs
to avoid the selection tools of the Zone entities to be considered as Children of it.
This was necessary for the display of the hierarchy status, but also for the "Add Children to Selection".

**For the QA:**
1- Open the "Create" Application
2- Make sure to have the "Hierarchy" column displayed in the entity list.
3- Create a new Zone entity
**Before this fix:**
The new zone entity was displayed as being a Top Parent. (because there was a local entity parented to it. (to display the grid box)
In In-world view, the bounding box was also indicating that it was a Top Parent (it was displayed in Orange instead of light gray)
If you selected that new zone, and do "**Selection... > Add Children to Selection**", the bounding box turned light gray as a multiple selection. If you were keeping CTRL pressed and deselect the zone only, then you could see the invisible local entity selected with a bounding box in cyan (as a child).

**After this fix:**
No more trace of any children, since local entities are now ignored when we get the list of the children.
So new zones are no more displayed as a Parent of a child that we didn't expect. And the  "**Selection... > Add Children to Selection**" won't consider this false child too.

### CODE REVIEW

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR874/Vircadia-Alpha-PR874-43f3ba5_22abf1c2.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR874/Vircadia-Alpha-PR874-43f3ba5_22abf1c2.exe

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR874/Vircadia-Alpha-PR874-43f3ba5_22abf1c2.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR874/Vircadia-Alpha-PR874-43f3ba5.tar_22abf1c2.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR874/Vircadia-Alpha-PR874-43f3ba5.tar_22abf1c2.gz

**Review by ctrlaltdavid — APPROVED** [review]

**Comment by kasenvr** [comment]
According to QA it worked as expected: I created the zone entity and it doesn't show any children under it on the icon. Though, I didn't test any parenting as I was unsure what causes what in that respect.

**Comment by AleziaKurdis** [comment]
Hmm... Can we use the create application to edit avatar entities?

Because we can't for "Local" entities. They are not visible in the list and can't be selected.
To be able to create a local entity or set it a local entities as child of another entity, the only way is to do it using a script.

**Comment by kasenvr** [comment]
We are able to adjust wearables with the Create App as far as I know.

**Comment by AleziaKurdis** [comment]
We will put this PR on HOLD.
I will change the strategy.
and figure to exclude those specific "tools" entities
instead of excluding local entities (and avatar entities)
Seems that eventually, Local entities could be visible like all the others.

**Comment by AleziaKurdis** [comment]
Finally, I change the approach.
I exclude now only the Entity Shape Visualizer 
instead of systematically all the local entities (and the avatar entities by the same time)

So this will continue to work for all types of entities. (domain, local, avatar)

I still don't think it will be viable to have the local entities visible in the create Application
it might seriously complexify everything to be able to protect the selection tool to self-edit

If we do this, I suggest that we implement a property to have specific local entities to never be returned by the findEntities### methods, getChildrens, and certainly some others.

**Comment by AleziaKurdis** [comment]
You can now remove the "**Do not Merge**" flag and gives this to QA.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR874/Vircadia-Alpha-PR874-4b9bf94_b633bc4a.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR874/Vircadia-Alpha-PR874-4b9bf94.tar_b633bc4a.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR874/Vircadia-Alpha-PR874-4b9bf94_b633bc4a.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR874/Vircadia-Alpha-PR874-4b9bf94.tar_b633bc4a.Z

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR874/Vircadia-Alpha-PR874-4b9bf94_b633bc4a.exe

File: scripts/system/create/edit.js [hunk 1]
```diff
@@ -2923,4 +2925,17 @@ function getParentState(id) {
     return state;
 }
 
+function getDomainOnlyChildrenIDs(id) {
+    var allChildren = Entities.getChildrenIDs(id);
+    var realChildren = [];
+    var properties;
+    for (var i = 0; i < allChildren.length; i++) {
+        properties = Entities.getEntityProperties(allChildren[i], ["name"]);
+        if (properties.name !== entityShapeVisualizerSessionName && properties.name !== undefined) {
```
Comment (ctrlaltdavid) [inline scripts/system/create/edit.js:2934]:
```suggestion
        if (properties.name !== undefined && properties.name !== entityShapeVisualizerSessionName) {
```
Thought it doesn't matter in this code, I suggest testing against `undefined` first, just as a matter of good habit.

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

**Review by ctrlaltdavid — APPROVED** [review]

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR874/Vircadia-Alpha-PR874-6b2ea04_343ded8c.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR874/Vircadia-Alpha-PR874-6b2ea04.tar_343ded8c.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR874/Vircadia-Alpha-PR874-6b2ea04.tar_343ded8c.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR874/Vircadia-Alpha-PR874-6b2ea04_343ded8c.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR874/Vircadia-Alpha-PR874-6b2ea04_343ded8c.exe

**Comment by Aitolda** [comment]
Seems to be working as expected now.

**Review by kasenvr — APPROVED** [review]
Works fine for me as well.

**Pull request merged**